x-pack/plugins/alerting/common/rule.ts

```diff
@@ -142,6 +142,11 @@ export interface RuleSystemAction {
   params: RuleActionParams;
   type: typeof RuleActionTypes.SYSTEM;
   useAlertDataForTemplate?: boolean;
+  // Workaround for typescript issues where boolean types are only defined in one of the union type members
+  // Allow frequency and alertsFilter to be deemed present in a system action, but require them to be undefined
+  // This prevents code like action.frequency?.someProperty from throwing an error
+  frequency?: undefined;
+  alertsFilter?: undefined;
```

Comment:
not a huge fan of this, wouldnt it be better to use type guards to determine which action it is?

Reply:
There are instances in this branch where we are using `!action.frequency` to determine if an action is a system action or not, because the `isSystemAction` type guard THROWS A TYPESCRIPT ERROR if you try to put something into it that isn't a default action

I can revert this change to get this merged but there's a lot of auditing we need to do on this feature branch to get the whole thing not to be spaghetti typedefs

Reply:
yea that makes sense. Im ok with it if there's an intention to fix this later.